modules/xmpp/ChatRoom.js

```diff
@@ -937,55 +935,36 @@ export default class ChatRoom extends Listenable {
      */
     onPresenceUnavailable(pres, from) {
         // ignore presence
-        if ($(pres).find('>ignore[xmlns="http://jitsi.org/jitmeet/"]').length) {
+        if ($_(pres, '>ignore[xmlns="http://jitsi.org/jitmeet/"]')) {
             return true;
         }
 
         // room destroyed ?
-        const destroySelect = $(pres).find('>x[xmlns="http://jabber.org/protocol/muc#user"]>destroy');
+        const destroySelect = $_(pres, '>x[xmlns="http://jabber.org/protocol/muc#user"]>destroy');
 
-        if (destroySelect.length) {
+        if (destroySelect) {
             let reason;
-            const reasonSelect
-                = $(pres).find(
-                    '>x[xmlns="http://jabber.org/protocol/muc#user"]'
-                        + '>destroy>reason');
+            const reasonSelect = $_(pres, '>x[xmlns="http://jabber.org/protocol/muc#user"]>destroy>reason');
 
-            if (reasonSelect.length) {
-                reason = reasonSelect.text();
+            if (reasonSelect) {
+                reason = reasonSelect.textContent;
             }
 
-            this.eventEmitter.emit(XMPPEvents.MUC_DESTROYED, reason, destroySelect.attr('jid'));
+            this.eventEmitter.emit(XMPPEvents.MUC_DESTROYED, reason, destroySelect.getAttribute('jid'));
             this.connection.emuc.doLeave(this.roomjid);
 
             return true;
         }
 
         // Status code 110 indicates that this notification is "self-presence".
-        const isSelfPresence
-            = $(pres)
-                .find(
-                    '>x[xmlns="http://jabber.org/protocol/muc#user"]>'
-                        + 'status[code="110"]')
-                .length;
-        const isKick
-            = $(pres)
-                .find(
-                    '>x[xmlns="http://jabber.org/protocol/muc#user"]'
-                        + '>status[code="307"]')
-                .length;
+        const isSelfPresence = Boolean($_(pres, '>x[xmlns="http://jabber.org/protocol/muc#user"]>status[code="110"]'));
+        const isKick = Boolean($_(pres, '>x[xmlns="http://jabber.org/protocol/muc#user"]>status[code="307"]'));
         const membersKeys = Object.keys(this.members);
 
         if (isKick) {
-            const actorSelect
-                = $(pres)
-                .find('>x[xmlns="http://jabber.org/protocol/muc#user"]>item>actor');
+            const actorSelect = $_(pres, '>x[xmlns="http://jabber.org/protocol/muc#user"]>item>actor');
 
-            let actorNick;
-
-            if (actorSelect.length) {
-                actorNick = actorSelect.attr('nick');
-            }
+            const actorNick = actorSelect?.getAttribute('nick') || undefined;
```

Comment:
Isn't the `|| undefined` redundant?

Reply:
I tried to not change functionality in any way. In the old code actorNick is defined without any value (thus undefined). 
However, I now see, that getAttribute (that potentially returns null and not undefined) will never get called if actorNick is null so yeah that can go